{{ cookiecutter.project_name | snakecase }}/CMakeLists.txt

```diff
@@ -31,18 +23,19 @@ endforeach()
 
 # Declare a C++ library
 add_library(
-    ${PROJECT_NAME} SHARED
-    src/{{ cookiecutter.class_name | snakecase }}.cpp
+  ${PROJECT_NAME}
+  src/{{ cookiecutter.class_name | snakecase }}.cpp
 )
 ament_target_dependencies(${PROJECT_NAME} ${THIS_PACKAGE_INCLUDE_DEPENDS})
 target_include_directories(${PROJECT_NAME}
-    PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
-           $<INSTALL_INTERFACE:include>)
+  PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+         $<INSTALL_INTERFACE:include>)
+target_compile_features(${PROJECT_NAME} PUBLIC cxx_std_17)
```

Comment:
Is there a way to let the user override this? I also wonder what's the difference between having `target_compile_features(${XXX} PUBLIC cxx_std_17)` for each target and global

```cmake
if(NOT CMAKE_CXX_STANDARD)
  set(CMAKE_CXX_STANDARD 17)
  set(CMAKE_CXX_STANDARD_REQUIRED ON)
endif()
```

Reply:
Do you think it's a good idea to have the standard version as a user input that default to 17? `cxx_std_{{ cookiecutter.cxx_version }}`